activesupport/lib/active_support/ordered_options.rb

```diff
@@ -92,18 +92,48 @@ def inspect
   #   h.boy  # => 'John'
   class InheritableOptions < OrderedOptions
     def initialize(parent = nil)
-      if parent.kind_of?(OrderedOptions)
+      @parent = parent
+      if @parent.kind_of?(OrderedOptions)
         # use the faster _get when dealing with OrderedOptions
-        super() { |h, k| parent._get(k) }
-      elsif parent
-        super() { |h, k| parent[k] }
+        super() { |h, k| @parent._get(k) }
+      elsif @parent
+        super() { |h, k| @parent[k] }
       else
         super()
+        @parent = {}
       end
     end
 
+    def to_h
+      @parent.dup.merge(self)
```

Comment:
```suggestion
      @parent.merge(self)
```

`merge` returns a copy already. Do I think this `dup` isn't needed, unless I'm missing something?